conn: use CmsgSpace() for ancillary data buf sizing

jwhited opened the pull request.

bradfitz commented on conn/conn.go:
```diff
@@ -0,0 +1,77 @@
+
+// A Bind listens on a port for both IPv6 and IPv4 UDP traffic.
+type Bind interface {
+	LastMark() uint32
```
docs on all these would be nice (including concurrency properties) if you know the answers.

bradfitz commented on conn/conn_linux.go:
```diff
@@ -111,47 +103,16 @@ func CreateEndpoint(s string) (Endpoint, error) {
-}
-
-func CreateBind(port uint16, device *Device) (*nativeBind, uint16, error) {
+func CreateBind(port uint16, device interface{}) (*nativeBind, uint16, error) {
```
exported func returning unexported concrete type is weird.

bradfitz replied:
and what's the returned uint16?

document?

crawshaw commented:
I figure each of these commits going upstream should leave things better than it was. I've tried to add some docs, but clearly there's more that could be done.

crawshaw commented:
(Anyway PTAL.)

crawshaw commented:
Moved to https://github.com/WireGuard/wireguard-go/pull/9.

jwhited committed:
conn: use CmsgSpace() for ancillary data buf sizing

CmsgLen() does not account for data alignment.

Reviewed-by: Adrian Dewhurst <adrian@tailscale.com>
Signed-off-by: Jordan Whited <jordan@tailscale.com>

jwhited committed:
conn: fix getSrcFromControl() iteration

We only expect a single control message in the normal case, but this
would loop infinitely if there were more.

Reviewed-by: Adrian Dewhurst <adrian@tailscale.com>
Signed-off-by: Jordan Whited <jordan@tailscale.com>
